src/components/activity/ActivityDirectiveChangelog.svelte

```diff
@@ -0,0 +1,365 @@
+
+  function formatDate(dateString: string) {
+    return new Date(dateString)
+      .toLocaleString(undefined, {
+        day: 'numeric',
+        hour: 'numeric',
+        minute: '2-digit',
+        month: 'numeric',
+        year: '2-digit',
+      })
+      .replace(',', '');
+  }
+
+  function formatParameterValue(parameterName: string, parameterValue: any) {
+    if (!activityType?.parameters || !activityType.parameters[parameterName]) {
+      return parameterValue;
+    }
+
+    const parameterType = activityType.parameters[parameterName].schema.type;
+    switch (parameterType) {
+      case 'duration':
+        return convertUsToDurationString(parameterValue);
+      default:
+        return parameterValue;
+    }
+  }
+
+  function diffArrays(itemName: string, current: any[], previous: any[], itemSchema: any) {
+    let differences: { [name: string]: { currentValue: any; previousValue: any } } = {};
+
+    for (let i = 0; i < current.length; i++) {
+      if (itemSchema.type === 'series') {
+        differences = {
+          ...differences,
+          ...diffArrays(`${itemName} > [${i}]`, current[i], previous[i], itemSchema.items),
+        };
+      } else if (typeof current[i].key !== 'undefined' && typeof current[i].value !== 'undefined') {
+        if (current[i].key !== previous[i].key) {
+          differences[`${itemName} > [${i}] > key`] = {
+            currentValue: current[i].key,
+            previousValue: previous[i].key,
+          };
+        }
+        if (current[i].value !== previous[i].value) {
+          differences[`${itemName} > [${i}] > value`] = {
+            currentValue: current[i].value,
+            previousValue: previous[i].value,
+          };
+        }
+      } else if (current[i] !== previous[i]) {
+        differences[`${itemName} > [${i}]`] = {
+          currentValue: current[i],
+          previousValue: previous[i],
+        };
+      }
+    }
+
+    return differences;
+  }
+
+  function diffRevisions(current: ActivityDirectiveRevision, previous: ActivityDirectiveRevision) {
+    let differences: { [name: string]: { currentValue: any; previousValue: any } } = {};
+
+    // Build list of all arguments that differ
+    Object.keys(current.arguments).forEach(argument => {
+      const schema = activityType?.parameters[argument].schema;
+
+      if (schema?.type === 'series') {
+        differences = {
+          ...differences,
+          ...diffArrays(argument, current.arguments[argument], previous.arguments[argument], schema.items),
+        };
+      } else if (current.arguments[argument] !== previous.arguments[argument]) {
+        differences[argument] = {
+          currentValue: current.arguments[argument],
+          previousValue: previous.arguments[argument],
+        };
+      }
+    });
+
+    // Manually check remaining fields that could have changed and require extra formatting
+
+    if (current.start_offset !== previous.start_offset) {
+      differences['Start Time'] = {
```

Comment:
One thing to ask users about might be time formatting here - do they care about the specific time values or are they more interested that the field changed? Have heard comments before that the time offset is not particularly understandable vs a more human readable format.
<img width="116" alt="image" src="https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/83d6ad32-aae0-49df-b2c2-c19b50b9a279">

Reply:
Taking another look, it looks like the designs specify displaying these in DOY format so I can update that. It is a good question though of which is ultimately more useful information.